lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@zeroedin-bill I was asking him for the psql -V output to confirm he was actually using PG 9.3 (because he's gone back/forth on what version he's using). The libpq version isn't really what matters either -- it's the value that PHP has for the PGSQL_LIBPQ_VERSION constant (considering they were suggesting that's the best solution to the problem, that's the value that matters).

@deeky666 has confirmed he was just guessing -- so yes, like I said -- using PGSQL_LIBPQ_VERSION to check what client version isn't a valid solution.


@riccardonar yes you are right -- the minor version doesn't matter.

@deeky666